UMAProject/Assets/UMA/Core/Extensions/DynamicCharacterSystem/Scripts/AssetItem.cs

```diff
@@ -82,64 +112,107 @@ public void CachSerializedItem()
 				// Let's search for it?
-				if (_SerializedItem == null)
+				if (itemObject == null)
 				{
 					string s = _Type.Name;
-					string[] guids = AssetDatabase.FindAssets(_Name+ " t:" + s);
+					string[] guids = AssetDatabase.FindAssets(_Name + " t:" + s);
 					if (guids.Length > 0)
 					{
 						_Guid = guids[0];
 						_Path = AssetDatabase.GUIDToAssetPath(_Guid);
-						_SerializedItem = AssetDatabase.LoadAssetAtPath(_Path, _Type);
+						itemObject = AssetDatabase.LoadAssetAtPath(_Path, _Type);
 					}
 				}
 			}
-			#endif
+#else
+			Object itemObject = null;
+#endif
+			return itemObject;
 		}
 
+		public void CacheSerializedItem()
+		{
+#if UNITY_EDITOR
+			if (_SerializedItem != null) return;
+#if SUPER_LOGGING
+            Debug.Log("Loading item in AssetItem: " + _Name);
+#endif
+			//if (IsAddressable) return;
+
+			_SerializedItem = GetItem();
+#endif
+        }
+
+        public static string GetEvilName(Object o)
+        {
+            if (!o)
+            {
+                return "<Not Found!>";
+            }
+            if (o is SlotDataAsset)
+            {
+                SlotDataAsset sd = o as SlotDataAsset;
+                return sd.slotName;
+            }
+            if (o is OverlayDataAsset)
+            {
+                OverlayDataAsset od = o as OverlayDataAsset;
+                return od.overlayName;
+            }
+            if (o is RaceData)
+            {
+                return (o as RaceData).raceName;
+            }
+
+            return o.name;
+        }
+
         public string EvilName
         {
             get
             {
                 Object o = Item;
-
-                if (!o)
-                {
-                    return "<Not Found!>";
-                }
-                if (o is SlotDataAsset)
-                {
-                    SlotDataAsset sd = o as SlotDataAsset;
-                    return sd.slotName;
-                }
-                if (o is OverlayDataAsset)
-                {
-                    OverlayDataAsset od = o as OverlayDataAsset;
-                    return od.overlayName;
-                }
-                if (o is RaceData)
-                {
-                    return (o as RaceData).raceName;
-                }
-
-                return o.name;
+                return GetEvilName(o);
             }
         }
-        #endregion
-        #region Methods (edit time)
-#if UNITY_EDITOR
+#endregion
+
+        public void AddReference()
+        {
+            ReferenceCount++;
+        }
 
-        public void ReleaseItem()
+        public void FreeReference()
         {
+            ReferenceCount = 0;
             _SerializedItem = null;
         }
 
-        public string ToString(string SortOrder)
+		public void ReleaseItem()
+		{
+            if (IsAddressable)
+            {
+                ReferenceCount--;
+                if (ReferenceCount < 1)
+                {
+                    FreeReference();
```

Comment:
Please toss a LogError if ReferenceCount goes negative. If people have bad resource management they should be warned.

Reply:
Error logged with:
                        Debug.LogError("Reference count is negative on AssetItem " + this._Name + " of Type " + this._TheType+". This should not happen.");